eng/pipelines/templates/steps/build-artifacts.yml

```diff
@@ -83,13 +90,42 @@ steps:
 
   - ${{ parameters.BeforePublishSteps }}
 
-  - publish: $(Build.ArtifactStagingDirectory)
-    artifact: packages 
-    condition: succeededOrFailed()
+  - pwsh: |
+      # handle where the parameter is undefined
+      $shouldPublishDocs = $true
+      $docParam = "${{ parameters.BuildDocs }}"
+      if ($docParam) {
+        # if we set it to false handle it here
+        $shouldPublishDocs = 	[System.Convert]::ToBoolean($docParam)
+      }
+      
+      $packagePublishes = ((Get-ChildItem '$(Build.ArtifactStagingDirectory)') | Measure-Object).Count
+      
+      if ($packagePublishes -eq 0) {
+          Write-Host "We are expecting to publish package artifacts, but no artifacts were found."
+          Write-Host (Get-ChildItem '$(Build.ArtifactStagingDirectory)')
```

Comment:
Wouldn't this always output nothing if nothing was found?

Reply:
Outside of weird powershell wonkiness, you're right.